service/history/historyEngine.go

```diff
@@ -894,8 +894,18 @@ func (e *historyEngineImpl) DescribeWorkflowExecution(ctx context.Context,
 				return nil, &workflow.InternalServiceError{Message: "Unable to get activity schedule event."}
 			}
 			p.ActivityType = scheduledEvent.ActivityTaskScheduledEventAttributes.ActivityType
-			p.LastStartedTimestamp = common.Int64Ptr(ai.StartedTime.UnixNano())
-			p.Attempt = common.Int32Ptr(ai.Attempt)
+			if state == workflow.PendingActivityStateScheduled {
+				p.ScheduledTimestamp = common.Int64Ptr(ai.ScheduledTime.UnixNano())
+			} else {
```

Comment:
Why is it else? Logically having both does make sense.

Reply:
I believe when the activity state is scheduled, the started time will be reset to 0. If it has already started, then the scheduled time is not important.

Reply:
If you are in PendingActivityStateScheduled will StartTime be set? Will it be set to the last attempt's start time? 

If StartedTime does have a meaningful value when you are in PendingActivityStateScheduled, then I agree we should return it from here always.

Reply:
In PendingActivityStateScheduled state, the startTime is always 0.

Reply:
In that case I am of the opinion that your existing implementation is correct.